pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -215,22 +229,35 @@ def on_load_checkpoint(self, checkpointed_state: Dict[str, Any]):
         self.best_model_score = checkpointed_state["best_model_score"]
         self.best_model_path = checkpointed_state["best_model_path"]
 
-    def save_checkpoint(self, trainer, pl_module):
+    def should_save(self, trainer, is_last=False):
+        epoch = trainer.current_epoch
+        global_step = trainer.global_step
+        should_save = not (
+            # negative conditions
+            self.save_top_k == 0 or
```

Comment:
About formatting: `black` is not strictly enforced but generally used in this project. The issue here is that you put your logical operators at the end whereas black puts them at the beginning. This means that this function will look quite different after somebody comes around and does automatic formatting.

Can you do it yourself? `black -S -l 120 model_checkpoint.py` and then fix the comment positions and other undesired changes in formatting

Reply:
Yeah, black was used before, but CI formatter had a different opinion. I'll figure it out

Reply:
The CI formatter probably complained about this: https://www.flake8rules.com/rules/W503.html

But the rule is outdated as mentioned in the link